openwisp_monitoring/device/migrations/0009_update_device_status_for_disabled_critical_checks.py

```diff
@@ -0,0 +1,33 @@
+from django.db import migrations
+
+from openwisp_monitoring.device.settings import get_critical_device_metrics
+
+
+def update_critical_device_metric_status(apps, schema_editor):
+    Check = apps.get_model(
+        'check', 'Check'
+    )  # Ensure 'device' is the app where model is.
+    DeviceMonitoring = apps.get_model('device_monitoring', 'DeviceMonitoring')
+
+    critical_metrics_keys = [metric['key'] for metric in get_critical_device_metrics()]
+
+    for check in Check.objects.filter(
+        is_active=False, check_type__in=critical_metrics_keys
+    ).iterator():
+        DeviceMonitoring.handle_critical_metric(check)
```

Comment:
the best will be to make sure we do not duplicate logic already being defined in `openwisp_monitoring/device/base/models.py`. Look for any similarities and create private methods in the models that can be used also here whenever possible.

Reply:
I believe this migration will fail, because `apps.get_model('device_monitoring', 'DeviceMonitoring')` does not return the real model, but a simulation generated from the Migration framework.

Try to rollback this migration, then set some critical checks to False in your dev env, then run the migration again. What happens? I think it's highly unlikely it will run properly.
We will likely have to load the real (aka "concrete") `DeviceMonitoring` model here.